kernel/Kconfig

```diff
@@ -862,25 +862,47 @@ config XIP
 
 menu "Security Options"
 
-config STACK_CANARIES
-	bool "Compiler stack canaries"
-	depends on ENTROPY_GENERATOR || TEST_RANDOM_GENERATOR
-	select NEED_LIBC_MEM_PARTITION if !STACK_CANARIES_TLS
+config NEED_STACK_CANARIES
+	bool
 	help
-	  This option enables compiler stack canaries.
+	  Hidden option to signal that software stack protection is enabled.
 
+choice
+	prompt "Stack canaries protection options"
+	optional
+	help
 	  If stack canaries are supported by the compiler, it will emit
 	  extra code that inserts a canary value into the stack frame when
 	  a function is entered and validates this value upon exit.
 	  Stack corruption (such as that caused by buffer overflow) results
 	  in a fatal error condition for the running entity.
-	  Enabling this option can result in a significant increase
-	  in footprint and an associated decrease in performance.
+	  Enabling this option, depending on the level chosen, can result in a
+	  significant increase in footprint and a corresponding decrease in performance.
 
 	  If stack canaries are not supported by the compiler an error
 	  will occur at build time.
 
-if STACK_CANARIES
+config STACK_CANARIES
+	bool "Compiler stack canaries for all functions"
```

Comment:
perhaps shorten the names as those are choice entries:
Current:
![image](https://github.com/user-attachments/assets/c7523824-95f8-42ba-9eb6-2cc08cb893c4)

Proposal:
![image](https://github.com/user-attachments/assets/b8f390bc-e087-4233-923a-b432a0158d0c)

```suggestion
	bool "All functions"
```
Only a proposal, thus non-blocking .

Reply:
Applied suggestions, much better indeed.